aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -274,22 +242,23 @@ static Tensor& prod_out(Tensor& result, const Tensor& self, IntArrayRef dim,
   return result;
 }
 
-static Tensor prod(const Tensor& self, IntArrayRef dim, bool keepdim, optional<ScalarType> dtype) {
+Tensor prod(const Tensor& self, int64_t dim, bool keepdim, c10::optional<ScalarType> dtype) {
   Tensor result;
-  native::prod_out(result, self, dim, keepdim, dtype);
+  native::prod_out_impl(result, self, dim, keepdim, dtype);
   return result;
 }
 
-Tensor prod(const Tensor &self, ScalarType dtype) {
-  return at::native::prod(self, {}, false, optional<ScalarType>(dtype));
+Tensor prod(const Tensor &self, c10::optional<ScalarType> dtype) {
+  Tensor result;
+  return at::native::prod_out_impl(result, self, {}, false, dtype);
 }
 
-Tensor prod(const Tensor &self) {
-  return at::native::prod(self, {}, false, c10::nullopt);
+Tensor& prod_out(Tensor& result, const Tensor& self, int64_t dim, bool keepdim, c10::optional<ScalarType> dtype) {
```

Comment:
same with prod.

Reply:
for prod it's just a static helper function, I don't see an unused one like there was for sum()

Reply:
err, nevermind, I'll look again, I see you were referencing another file with some of these comments.

Reply:
yeah I think this is correct, looks like your other comments were referencing the BC issue. Here for prod we just have three functions and one static helper.